### ISSUE TITLE
[Terraform] AWS Auto Scaling Group With No Associated ELB False Positive

### ISSUE THREAD

**Issue by mariomsilva** [Checkmarx/kics#5547]
### Expected Behavior
Not trigger an Auto Scaling Group With No Associated ELB when `load_balancers` is not defined. The `load_balancers` is an optional attribute and should not be defined for Application Load Balancers as described in documentation:

> `load_balancers` (Optional) A list of elastic load balancer names to add to the autoscaling group names. Only valid for classic load balancers. For ALBs, use target_group_arns instead.
[Source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#load_balancers)

### Actual Behavior

```shell
Auto Scaling Group With No Associated ELB, Severity: MEDIUM, Results: 1
Description: AWS Auto Scaling Groups must have associated ELBs to ensure high availability and improve application performance. This means the attribute 'load_balancers' must be defined and not empty.
Platform: Terraform
	[1]: modules/autoscaling/main.tf:34
		033: 
		034: resource "aws_autoscaling_group" "my_asg" {
		035:   name_prefix          = format("%s-", var.name)
```

### Steps to Reproduce the Problem

```terraform
resource "aws_autoscaling_group" "my_asg" {
  name_prefix          = format("%s-", var.name)
  vpc_zone_identifier  = var.private_zone_identifiers
  launch_configuration = aws_launch_configuration.config.name
  target_group_arns    = var.target_group_arns
  min_size             = 1
  max_size             = 2
  desired_capacity     = 1
  instance_refresh {
    strategy = "Rolling"
    preferences {
      min_healthy_percentage = 50
    }
  }
}
```

**Comment by cxMiguelSilva** [comment]
Hi, @mariomsilva hope you are doing great!
Thank you for your input.
Indeed it was needed to check the use of ALB's once it was determined that ELB were not in use.
To solve this issue changes were made to the Security Query.
Can we add your code sample to the query sample test files?

**Comment by kaplanlior** [comment]
> Hi @cxMiguelSilva , Sure, nothing to worry about.
Thanks @mariomsilva. Can you confirm that we can use the examples under Apache 2.0 license ?

**Comment by mariomsilva** [comment]
Hi @kaplanlior,

100%. Is a sample extracted from my personal projects to test KICS. But if you wish, I can create exclusive new samples for you, based on PR  #5593, without variables and more hard code strings. As you wish!

**Comment by mariomsilva** [comment]
Sample based on the naming convention used in this project.

```terraform
resource "aws_autoscaling_group" "foo" {
  name_prefix          = 'bar-'
  vpc_zone_identifier  = ['subnet-abcd1234', 'subnet-1a2b3c4d']
  launch_configuration = aws_launch_configuration.foobar.name
  target_group_arns    = ['bar', 'baz', 'qux']
  min_size             = 1
  max_size             = 3
  desired_capacity     = 2
  instance_refresh {
    strategy = "Rolling"
    preferences {
      min_healthy_percentage = 50
    }
  }
}
```

**Comment by mariomsilva** [comment]
> Hi, @mariomsilva hope you are doing great! Thank you for your input. Indeed it was needed to check the use of ALB's once it was determined that ELB were not in use. To solve this issue changes were made to the Security Query.
#5593

I'm not an expert in Open Policy Agent language, so please correct me if I'm wrong. My understanding is you excluded the `target_group_arns` from the Auto Scaling Group with no associated ELB validation policy. I don't think `target_group_arns` should be ignored from the validation. The correct approach should be to verify the presence of `load_balancers` **or** `target_group_arns`. [Elastic Load Balancing](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb#load_balancer_type) (ELB) are Application Load Balancer(`application`), Gateway Load Balancer(`gateway`) and Network Load Balancer(`network`).

**Comment by cxMiguelSilva** [comment]
Nothing was excluded from the Rego Policy. Added an additional check to ensure that results are only presented to the user once neither load_balancers nor target_group_arns are set.
If you have any other questions let me know!
